transaction-exclusion-api/app/src/integrationTest/kotlin/net/consensys/linea/transactionexclusion/TransactionExclusionAppTest.kt

```diff
@@ -0,0 +1,428 @@
+package net.consensys.linea.transactionexclusion
+
+import com.sksamuel.hoplite.Masked
+import io.restassured.RestAssured
+import io.restassured.builder.RequestSpecBuilder
+import io.restassured.http.ContentType
+import io.restassured.response.Response
+import io.restassured.specification.RequestSpecification
+import io.vertx.core.json.JsonObject
+import io.vertx.junit5.VertxExtension
+import kotlinx.datetime.Clock
+import net.consensys.encodeHex
+import net.consensys.linea.async.get
+import net.consensys.linea.transactionexclusion.TransactionExclusionServiceV1.SaveRejectedTransactionStatus
+import net.consensys.linea.transactionexclusion.app.AppConfig
+import net.consensys.linea.transactionexclusion.app.DatabaseConfig
+import net.consensys.linea.transactionexclusion.app.DbCleanupConfig
+import net.consensys.linea.transactionexclusion.app.DbConnectionConfig
+import net.consensys.linea.transactionexclusion.app.PersistenceRetryConfig
+import net.consensys.linea.transactionexclusion.app.TransactionExclusionApp
+import net.consensys.linea.transactionexclusion.app.api.ApiConfig
+import net.consensys.linea.transactionexclusion.test.defaultRejectedTransaction
+import net.consensys.toHexString
+import net.consensys.trimToMillisecondPrecision
+import net.consensys.zkevm.persistence.db.DbHelper
+import net.consensys.zkevm.persistence.db.test.CleanDbTestSuiteParallel
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import java.time.Duration
+import kotlin.random.Random
+
+@ExtendWith(VertxExtension::class)
+class TransactionExclusionAppTest : CleanDbTestSuiteParallel() {
```

Comment:
Here, I'm making this integration test to be able to run each test case in parallel with no order restriction instead of in sequential order (which was the initial design), the tradeoff is more databases to create (each per test case) and more code for initial setup in each test case.